### PR TITLE
feat: add Heart theme option to Elm Modal and EmptyState (for Heart illustrations)

### PR DESCRIPTION
**Pull request by mbylstra**
# Summary
- Adds Heart theme option to the Elm Modal and Empty State  illustrations (cautionary, success, informative, negative, action). Defaults to Zen if not hooked up to a "prop drilling" solution of passing the react `themeKey` to the component.
- Adds elm code to @kaizen/design-tokens for themes. Includes `Theme` type (`Zen` or `Heart`) that needs to be passed down to the components. Also a `themeDecoder` for passing the theme to the Elm app via React.
- Exposes an `image` helper function/component in `Kaizen.HostedAssets.Image` to create images that use a path from our hosted assets S3 bucket (we are currently not using hosted assets anywhere in Kaizen for Elm yet, so this was required to use the new illustrations)

#### other things:
- I upgraded the usages of zen illustrations in Elm modal to use the ones in hosted-assets rather than the ones in this repo. I think there is now an opportunity to delete them.
- The stories always show the Heart theme, because the Elm storybook doesn't support theme switching (it's possible, but out of scope for this PR)
- The only other component I could think of with illustrations was Guideance block, but there is no Elm GuideanceBlock component.
- Supporting illustration animation is out of scope.

# Examples:

image

image

# Story links: 
https://dev.cultureamp.design/elm-modal-with-heart-illustrations/storybook/?path=/story/emptystate-elm--action
https://dev.cultureamp.design/elm-modal-with-heart-illustrations/storybook/?path=/story/modal-elm--confirmation-cautionary-shown-by-default

# Checklist

- [ ] If this contains visual changes, has it been reviewed by a designer?
- [ ] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [ ] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

**Comment by mbylstra** [comment]
I will wait for approval from a designer, to make sure this compromise is acceptable.

File: packages/hosted-assets/Kaizen/HostedAssets.elm
Comment (mbylstra) [inline packages/hosted-assets/Kaizen/HostedAssets.elm:27]: The code that determines how the `img` is rendered is based on https://github.com/cultureamp/kaizen-design-system/blob/master/packages/component-library/components/Icon/Icon.elm. Only the `Presentation` variant has been implemented. The `Img` variant can be added if a need arises.

File: packages/hosted-assets/Kaizen/HostedAssets/Image.elm
Comment (mbylstra) [inline packages/hosted-assets/Kaizen/HostedAssets/Image.elm:10]: This just supports presentation for now (completely hidden from screen readers), but other types where you want a description to be read to a screen reader could be added.

File: packages/hosted-assets/Kaizen/HostedAssets/Image.elm
Comment (mbylstra) [inline packages/hosted-assets/Kaizen/HostedAssets/Image.elm:70]: This just supports presentation for now (completely hidden from screen readers), but other types where you want a description to be read to a screen reader could be added.

**Comment by mbylstra** [comment]
[Kaizen: support Heart illustrations for EmptyState and Modal](https://trello.com/c/9m0ArRF9/1040-kaizen-support-heart-illustrations-for-emptystate-and-modal)

File: packages/design-tokens/elm/Kaizen/Theme.elm
Comment (sentience) [inline packages/design-tokens/elm/Kaizen/Theme.elm:28]: Any reason for this fallback? I wonder if the decoder should `fail` when it receives an unexpected string value, and if we want to tolerate a missing/invalid value then we can use `Decode.maybe` for that:

```elm
decodeValue (maybe (field "theme" themeDecoder)) flags
```

**Review by sentience — APPROVED** [review]
Looks good! Left a comment with one idea for improvement, but I don't consider it a blocker.

File: packages/design-tokens/elm/Kaizen/Theme.elm
Comment (mbylstra) [reply packages/design-tokens/elm/Kaizen/Theme.elm:28]: Thanks for the review Kevin!

I was a bit torn on this. I definitely didn't want it to fail, because I'm not very confident this would  be handled well by consumers. Falling back the last theme (in this case Zen) seems better than the page crashing or showing a generic error page. Also, I just wanted to make this as easy as possible for consumers (eg, not having to handle Maybes), as there will be a lot of friction in using the theme switcher in Elm so I'd be more worried people wouldn't find the time to implement it at all (which would guarantee no theme switching!). Even with these new features in Kaizen, there's still a lot of work to do to hook up the new illustrations - using the React `ThemeContext`, pass a flag to Elm and then prop drilling the theme all that way down to every Elm component. I happen to have a good amount of time this week so I can go deep with the prop drilling, etc, to get in our app but I'm not sure I can expect people in other teams with Elm codebases to do the same. 

Earlier as a much easier approach I suggested just immediately upgrading all the Elm illustrations to Heart, which would mean some customers not on the feature flag would see some components with Zen styles but Heart illustrations (because I thought implementing theme switching in Elm would be too much work for people). The Heart rollout team wasn't comfortable with this idea though.

There might be a much more magical solution for doing something like "context" in Elm to magically pass vars down the chain, but that would require a lot more investigation of feasibility.

**Pull request merged**
